library/Vanilla/Formatting/DateTimeFormatter.php

```diff
@@ -218,14 +218,18 @@ public function formatSeconds(int $seconds): string {
      * @param string $dateTime The Mysql-formatted datetime to convert to a timestamp. Should be in one
      * of the following formats: YYYY-MM-DD or YYYY-MM-DD HH:MM:SS.
      * @param mixed $fallback The value to return if the value couldn't be properly converted.
-     * @return int A timestamp or now if it couldn't be parsed properly.
+     * @param mixed $emptyFallback The fallback for an empty value. If not supplied then the `$fallback` will be used.
+     * @return int|null A timestamp or now if it couldn't be parsed properly.
      */
-    public static function dateTimeToTimeStamp(?string $dateTime = '', $fallback = null): int {
-        if (($testTime = strtotime($dateTime)) !== false) {
+    public static function dateTimeToTimeStamp(?string $dateTime, $fallback = false, $emptyFallback = false): ?int {
```

Comment:
We did go from making the first parameter optional, to making it required. Was this intentional?

Reply:
That was intentional, yes. It makes no sense to supply no arguments to a method that's entire purpose is to operate on the first parameter. The method as it was before also had a default that was a warning.

Reply:
As a side note. I don't see any calls that don't supply any arguments across our codebase.